peer/config.go

```diff
@@ -0,0 +1,228 @@
+	"github.com/lightningnetwork/lnd/discovery"
+	"github.com/lightningnetwork/lnd/htlcswitch"
+	"github.com/lightningnetwork/lnd/htlcswitch/hodl"
+	"github.com/lightningnetwork/lnd/htlcswitch/hop"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/invoices"
+	"github.com/lightningnetwork/lnd/lnpeer"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/netann"
+	"github.com/lightningnetwork/lnd/pool"
+	"github.com/lightningnetwork/lnd/queue"
+	"github.com/lightningnetwork/lnd/watchtower/wtclient"
+)
+
+// Config defines configuration fields that are necessary for a peer object
+// to function.
+type Config struct {
+	// Conn is the underlying network connection for this peer.
+	Conn net.Conn
+
+	// ConnReq stores information related to the persistent connection request
+	// for this peer.
+	ConnReq *connmgr.ConnReq
+
+	// PubKeyBytes is the serialized, compressed public key of this peer.
+	PubKeyBytes [33]byte
+
+	// Addr is the network address of the peer.
+	Addr *lnwire.NetAddress
+
+	// Inbound indicates whether or not the peer is an inbound peer.
+	Inbound bool
+
+	// Features is the set of features that we advertise to the remote party.
+	Features *lnwire.FeatureVector
+
+	// LegacyFeatures is the set of features that we advertise to the remote
+	// peer for backwards compatibility. Nodes that have not implemented
+	// flat features will still be able to read our feature bits from the
+	// legacy global field, but we will also advertise everything in the
+	// default features field.
+	LegacyFeatures *lnwire.FeatureVector
+
+	// OutgoingCltvRejectDelta defines the number of blocks before expiry of
+	// an htlc where we don't offer it anymore.
+	OutgoingCltvRejectDelta uint32
+
+	// ChanActiveTimeout specifies the duration the peer will wait to request
+	// a channel reenable, beginning from the time the peer was started.
+	ChanActiveTimeout time.Duration
+
+	// ErrorBuffer stores a set of errors related to a peer. It contains error
+	// messages that our peer has recently sent us over the wire and records of
+	// unknown messages that were sent to us so that we can have a full track
+	// record of the communication errors we have had with our peer. If we
+	// choose to disconnect from a peer, it also stores the reason we had for
+	// disconnecting.
+	ErrorBuffer *queue.CircularBuffer
+
+	// WritePool is the task pool that manages reuse of write buffers. Write
+	// tasks are submitted to the pool in order to conserve the total number of
+	// write buffers allocated at any one time, and decouple write buffer
+	// allocation from the peer life cycle.
+	WritePool *pool.Write
+
+	// ReadPool is the task pool that manages reuse of read buffers.
+	ReadPool *pool.Read
+
+	// Switch is a pointer to the htlcswitch. It is used to setup, get, and
+	// tear-down ChannelLinks.
+	Switch *htlcswitch.Switch
```

Comment:
In the future, we can abstract this with a _slimmer_ interface. Same goes for the other raw pointers below.